src/phoenix/server/api/types/Span.py

```diff
@@ -133,17 +143,24 @@ def descendants(
         if (traces := info.context.traces) is None:
             return []
         return [
-            to_gql_span(cast(trace_schema.Span, traces[span_id]))
+            to_gql_span(cast(trace_schema.Span, traces[span_id]), info.context.evals)
             for span_id in traces.get_descendant_span_ids(
                 cast(SpanID, self.context.span_id),
             )
         ]
 
 
-def to_gql_span(span: trace_schema.Span) -> "Span":
+def to_gql_span(span: trace_schema.Span, evals: Optional[Evals] = None) -> "Span":
     events: List[SpanEvent] = list(map(SpanEvent.from_event, span.events))
     input_value = cast(Optional[str], span.attributes.get(INPUT_VALUE))
     output_value = cast(Optional[str], span.attributes.get(OUTPUT_VALUE))
+    span_evaluations: List[SpanEvaluation] = []
+    document_evaluations: List[DocumentEvaluation] = []
+    span_id = span.context.span_id
+    for evaluation in evals.get_evaluations_by_span_id(span_id) if evals else ():
+        span_evaluations.append(SpanEvaluation.from_pb_evaluation(evaluation))
+    for evaluation in evals.get_document_evaluations_by_span_id(span_id) if evals else ():
+        document_evaluations.append(DocumentEvaluation.from_pb_evaluation(evaluation))
```

Comment:
optimization: You can place this code of getting evaluations and documents on the Span node. This will alleviate the load of the query if the query doesn't ask for anything related to evals. If there are multiple fields that require evaluations by span_id, you can wrap that in a dataloader to eliminate the n+1. Can do this as a follow-up but it's a worthwhile refactor as it's always good to eliminate over-fetching if possible.

Reply:
Some links to dataloading: https://leebyron.com/dataloader-v2/

Reply:
oh yea, that's right. we had talked about this before. I totally forgot about it